modules/javafx.controls/src/test/java/test/javafx/scene/control/TreeTableRowTest.java

```diff
@@ -820,4 +834,71 @@ public class TreeTableRowTest {
         TreeTableRow cell = new TreeTableRow();
         cell.setSkin(new TreeTableRowSkin(cell));
     }
+
+    /** TreeTableView with cell selection enabled should not select TreeTableRows */
+    @Test
+    public void test_TreeTableView_jdk_8292353_select_all() {
+        TreeTableView<String> tree = ControlUtils.createTreeTableView();
+
+        stageLoader = new StageLoader(tree);
+        TreeTableView.TreeTableViewSelectionModel<String> sm = tree.getSelectionModel();
+        sm.setSelectionMode(SelectionMode.MULTIPLE);
+        sm.setCellSelectionEnabled(true);
+        sm.clearSelection();
+
+        TreeTableColumn<String,?> col0 = tree.getColumns().get(0);
+        TreeTableColumn<String,?> col1 = tree.getColumns().get(1);
+        TreeTableColumn<String,?> col2 = tree.getColumns().get(2);
+        TreeTableRow row = ControlUtils.getTreeTableRow(tree, 0);
+        TreeTableCell c0 = ControlUtils.getTreeTableCell(tree, 0, 0);
+        TreeTableCell c1 = ControlUtils.getTreeTableCell(tree, 0, 1);
+        TreeTableCell c2 = ControlUtils.getTreeTableCell(tree, 0, 2);
+
+        assertFalse(c0.isSelected());
+        assertFalse(c1.isSelected());
+        assertFalse(c2.isSelected());
+        assertFalse(row.isSelected());
+
+        // select all cells in the first row
+        sm.select(0, col0);
+        sm.select(0, col1);
+        sm.select(0, col2);
+
+        assertTrue(c0.isSelected());
+        assertTrue(c1.isSelected());
+        assertTrue(c2.isSelected());
+        assertFalse(row.isSelected()); // JDK-8292353 failure
+    }
+
+    /** TreeTableView with cell selection enabled should not select TreeTableRows */
+    @Test
+    public void test_TreeTableView_jdk_8292353_select_all_but_one() {
```

Comment:
This test passes with or without the code changes.
If test intention is to increase coverage, then the comment `// JDK-8292353 failure` needs to be removed. If not, the test needs some correction.

Reply:
I don't understand.  It fails for me

```
TreeTableRowTest > test_TreeTableView_jdk_8292353_select_all_but_one FAILED
    org.opentest4j.AssertionFailedError: expected: <false> but was: <true>
        at app//org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at app//org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:40)
        at app//org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:35)
        at app//org.junit.jupiter.api.Assertions.assertFalse(Assertions.java:227)
        at app//test.javafx.scene.control.TreeTableRowTest.test_TreeTableView_jdk_8292353_select_all_but_one(TreeTableRowTest.java:902)
```

command line:

```
/bin/sh gradlew -PSTUB_RUNTIME=../caches/modular-sdk -PFULL_TEST=true -PUSE_ROBOT=true --no-daemon --info --stacktrace cleanTest :controls:test --tests TreeTableRowTest -x :web:test --stacktrace 2>&1 | tee ~/`date +"test-%Y-%m%d-%H%M%S"`.log
```

test and ControlUtils supplanted on top of the current master.

did you switch and rebuild?

Reply:
A clean build and run behaves correctly. This test fails without code changes and passes with them.
Sorry for the confusion!

Reply:
Thank you, Ajit!